packages/transformer/src/IModelTransformer.ts

```diff
@@ -606,13 +612,69 @@ export class IModelTransformer extends IModelExportHandler {
     return this._cachedSynchronizationVersion;
   }
 
+  /** the changeset in the scoping element's source version found for this transformation
+   * @note: the version depends on whether this is a reverse synchronization or not, as
+   * it is stored separately for both synchronization directions.
+   * @note: empty string and -1 for changeset and index if it has never been transformed, or was transformed before federation guid update (pre 1.x).
+   */
+  protected get synchronizationVersion(): ChangesetIndexAndId {
+    if (this._cachedSynchronizationVersion) {
+      return this._cachedSynchronizationVersion;
+    }
+
+    const provenanceScopeAspect = this.tryGetProvenanceScopeAspect();
+    if (!provenanceScopeAspect) {
+      return { index: -1, id: "" }; // first synchronization.
+    }
+
+    const version = this._options.isReverseSynchronization
+      ? (JSON.parse(provenanceScopeAspect.jsonProperties ?? "{}") as TargetScopeProvenanceJsonProps).reverseSyncVersion
+      : provenanceScopeAspect.version;
+    if (!version) {
+      return { index: -1, id: "" }; // previous synchronization was done before fed guid update.
+    }
+
+    const [id, index] = version.split(";");
+    this._cachedSynchronizationVersion = { index: Number(index), id };
+    return this._cachedSynchronizationVersion; // synchronization version found and cached.
+  }
+
+  /**
+   * @returns provenance scope aspect if it exists in the provenanceDb.
+   * Provenance scope aspect is created and inserted into provenanceDb when [[initScopeProvenance]] is invoked.
+   */
+  protected tryGetProvenanceScopeAspect(): ExternalSourceAspect | undefined {
```

Comment:
are there any places we can use this in the transformer today where we need this info?

Reply:
Yes. We update this aspect in several cases.
1. If we run `processChanges()` on an old branch (that was created before fedGuid update), we fake the version data on the Scope aspect ourselves in order to make the transformer resolve the correct startChangesetIndex.
2. We need to increment reverse sync version by 1 after a reverse synchronization because transformer doesn't take into account that the changeset will be pushed after the transformation on a source iModel. This is necessary for partial fork support.

Reply:
I actually meant within the core-transformer.

Do you think you could contribute your implementation of "upgrading" an old branching relationship? Assuming it's not specific to "partial forks"

Reply:
@nick4598 is going to be writing a test of "upgrading" an old branching relationship and your implementation could be helpful

Reply:
It doesn't have to be in this PR but we'd appreciate it, we were planning to add that separately